samples: webusb: Host webusb  demo app directly on zephyr doc

finikorg opened the pull request:
Move artifacts from private repository to zephyr docs.

Fixes: https://github.com/finikorg/webusb-sample/issues/3

kartben committed:
samples: webusb: Include demo.rst to documentation

Include demo.rst to the webusb sample documentation.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

finikorg committed:
doc: samples: usb: Include only README to TOC

Include only README.rst to docs, skipping helpers included with :doc:
directive.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: webusb: Copy needed file from the external repository

Copy file index.html from the external repository
finikorg/webusb-sample.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
samples: webusb: Update README

Update webusb README with recent changes after index.html was moved to
sample from the external repository.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

kartben commented on samples/subsys/usb/webusb/README.rst:
```diff
@@ -37,15 +37,15 @@ Build and flash webusb sample with:
+*********************************************
 
-This sample application requires latest Google Chrome, a web page
+This sample application requires the latest Google Chrome, a web page
```
would it make sense to reword along the lines of ".... a modern browser, supporting WebUSB <link to https://caniuse.com/webusb>" ? Since Edge is also supported, it might make sense to be slightly more broad here.

finikorg replied:
Is this https://caniuse.com/ credible enough to be put in the docs?

kartben replied:
ya good point. I think it's trustworthy but we shouldn't necessarily link to a 3rd party unless really needed. It's fine to stick to "latest Chrome" wording, I guess.

kartben reviewed (dismissed):
That's awesome - thanks!
Consider my only comment as non-blocking.

Note: the demo page is effectively `<html>` within `<html>` which is probably a bit hackish and not really w3c compliant, but I'd say the tradeoff is worthwhile (plus, people have the link to the raw source if needed).

finikorg commented on samples/subsys/usb/webusb/README.rst:
```diff
@@ -78,7 +79,8 @@ Follow these steps to run the demo on your host system:
    in Chrome on Windows. See https://github.com/WICG/webusb#implementation-status
 
-#. Click on Connect button to connect to the device.
+#. Click on the :guilabel:`Connect` button to connect to the device.
```
Yes, found this myself today only:)

kartben reviewed (dismissed):
While I appreciate that you kept commit authorship set to myself for the first commit, it doesn't look right as it probably doesn't compile as-is. Maybe just squash the first two? Or, probably better, re-order with a first commit that moves "your" code to clearly show it's you that's contributing this, and then do the rest with co-authored-by in a second commit.
Thanks again!

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+</script>
+ <button id="connect" style="visibility: initial">Connect To WebUSB Device</button>
+ <br><br><label for="title">Sender: </label> <br>
+ <textarea id="editor", rows="25" cols="80" id="source">WebUSB!</textarea>
```
```suggestion
 <textarea id="input" rows="25" cols="80">WebUSB!</textarea>
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+ <br><button id="submit">Send</button>
+ <br><br>
+ <label for="title">Receiver: </label> </br>
+ <textarea id="output", rows="25" cols="80" id="source"></textarea>
```
```suggestion
 <textarea id="output" rows="25" cols="80"></textarea>
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+ <textarea id="editor", rows="25" cols="80" id="source">WebUSB!</textarea>
+ <br><button id="submit">Send</button>
+ <br><br>
+ <label for="title">Receiver: </label> </br>
```
```suggestion
 <label for="title">Receiver: </label> <br>
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+<html>
```
```suggestion
<!DOCTYPE html>
<html>
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+
+</script>
+ <button id="connect" style="visibility: initial">Connect To WebUSB Device</button>
+ <br><br><label for="title">Sender: </label> <br>
```
```suggestion
 <br><br><label for="input">Sender: </label> <br>
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+  }
+
+  document.querySelector("#submit").onclick = () => {
+    let source = document.querySelector("#editor").value;
```
```suggestion
    let source = document.querySelector("#input").value;
```

kartben commented on samples/subsys/usb/webusb/index.html:
```diff
@@ -0,0 +1,125 @@
+ <br><br>
+ <label for="title">Receiver: </label> </br>
+ <textarea id="output", rows="25" cols="80" id="source"></textarea>
+</body>
```
missing closing `</html>`

```suggestion
</body>
</html>
```

kartben reviewed (commented):
I spotted a few other issues in the HTML document by putting it through a validator, could you please have a look

kartben committed:
samples: webusb: Fix HTML validator issues

Running the HTML code through W3C Validator revealed several issues with
the markup that this commit fixes.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

finikorg reviewed (commented):
ack @kartben commits
